src/main/java/core/basesyntax/data/DataConverterImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.data;
+
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.List;
+import java.util.Objects;
+
+public class DataConverterImpl implements DataConverter {
+    private Formater formater;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(s -> {
+                    if (formater == null) {
+                        formater = new FormaterImpl(s);
+                        return null;
```

Comment:
Returning null within a stream map operation is not a good practice. It would be better to handle the initialization of the 'formater' before streaming over the inputReport.